plugins/kvscheduler/api/txn_record.go

```diff
@@ -172,6 +172,11 @@ type RecordedTxnOps []*RecordedTxnOp
 // RecordedTxns is a list of recorded transactions.
 type RecordedTxns []*RecordedTxn
 
+type RecordedKVWithMetadata struct {
```

Comment:
Just add some explaining godoc for this new structure. When i dig deeper for the difference between this and KVWithMetadata, i got only added ProtoMsgName field (in RecordedProtoMessage struct somewhere in internal). So please state clearly why we need it, what is the added value (+maybe compare with KVWithMetadata).